db/db.go

```diff
@@ -85,6 +85,38 @@ func RunInTransactionWithResult[T any](ctx context.Context, dbConnectionPool DBC
 	return result, nil
 }
 
+// RunInTransactionWithPostCommit runs the given atomic function in an atomic database transaction.
+// If the atomic function succeeds, it returns a postCommit function to be executed after the transaction is committed.
+func RunInTransactionWithPostCommit(ctx context.Context, dbConnectionPool DBConnectionPool, opts *sql.TxOptions, atomicFunction func(dbTx DBTransaction) (func() error, error)) error {
```

Comment:
The method signature and method usage could become simpler if you update the method signature to receive a a struct with function like parameters, something like this:

```go
func RunInTransaction(ctx context.Context, opts DBTransactionOptions) error {
	dbConnectionPool := opts.DBConnectionPool
	atomicFunction := opts.AtomicFn
	postCommitFn := opts.PostCommitFn
	sqlOpts := opts.SQLOptions
	//...
```

Reply:
That's a good simplification 👍